CONTRIBUTING.md

```diff
@@ -0,0 +1,94 @@
+
+Before contributing, we encourage you to read our CONTRIBUTING policy (you are here), our [LICENSE](LICENSE), and our [README](README.md), all of which should be in this repository.
+
+## Issues ##
+
+If you want to report a bug or request a new feature, the most direct method is to [create an issue](https://github.com/AbanteAI/mentat/issues/new) in this repository.  We recommend that you first search through existing issues (both open and closed) to check if your particular issue has already been reported.  If it has then you might want to add a comment to the existing issue.  If it hasn't then feel free to create a new one.
+
+## Pull requests ##
+
+If you choose to [submit a pull request](https://github.com/AbanteAI/mentat/pulls), you will notice that our continuous integration (CI) system runs a fairly extensive set of linters and syntax checkers.  Your pull request may fail these checks, and that's OK.  If you want you can stop there and wait for us to make the necessary corrections to ensure your code passes the CI checks.
+
+If you want to make the changes yourself, or if you want to become a regular contributor, then you will want to set up [pre-commit](https://pre-commit.com/) on your local machine.  Once you do that, the CI checks will run locally before you even write your commit message.  This speeds up your development cycle considerably.
+
+## Getting Started
+
+### Contribution Process.
+
+Mentat follows the usual open source contribution process on Github. In order to contribute, you must fork the repository, create a branch based on the main branch, make your changes there and then create a pull request. Github has excellent documentation (and contributed documentation and videos) on how to do this. See [Contributing to a project](https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project), [first contributions](https://github.com/firstcontributions/first-contributions) for more details.
+
+The project, like most projects, does have some customizations specific to Mentat such as code formatting, quality checks which are enforced on each pull request using Github actions. Pull Requests (PRs - described later) are not accepted unless all the checks are passed.
+
+### Setup the Environment.
+
+Mentat requires Python (version 3.10 or later) for development. The source code comes with two sets of requirements. The first in requirements.txt are needed to run Mentat. The second additional requirements in dev-requirements.txt are needed to do development work. As a contributor you will need to install both sets of requirements to have the full set of tools. In addition you will need some tools outside of the python environment. 
+
+#### Python Virtual Environment setup
```

Comment:
I think it's best for this and the previous bullet to have a code block. Also it might be good to have a section on how to run the tests and other automatic checks you might want to run before pushing black/ruff/isort/pyright.

Reply:
Made the changes